arangod/Agency/Agent.cpp

```diff
@@ -1968,8 +1970,20 @@ void Agent::emptyCbTrashBin() {
 
   LOG_TOPIC("12ad3", DEBUG, Logger::AGENCY) << "unobserving: " << envelope->toJson();
 
-  // Best effort. Will be retried anyway
-  auto wres = write(envelope);
+  // This is a best effort attempt. If either the queueing or the write fail,
+  // while above _callbackTrashBin has been cleaned, entries will repopulate with
+  // future 404 errors, when they are triggered again. So either way these attempts
+  // are repeated until such time, when the callbacks are gone successfully through
+  // queue + write.
+  auto* scheduler = SchedulerFeature::SCHEDULER;
+  if (scheduler != nullptr) {
+    scheduler->queue(RequestLane::INTERNAL_LOW, [envelope = std::move(envelope)] {
```

Comment:
not checking the return value here will trigger a compile warning in some build environments, which hopefully makes the build fail.
Please check the return value here, even if you plan to ignore it.

Reply:
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/5930/